src/formatMarkup.js

```diff
@@ -39,28 +68,30 @@ export const diffableFormatter = function (markup) {
     let result = '\n' + '  '.repeat(indent) + '<' + node.nodeName;
 
     // Add attributes
-    if (node?.attrs?.length === 1) {
-      let attr = node?.attrs[0];
-      result = result + ' ' + attr.name + '="' + attr.value + '">';
-    } else if (node?.attrs?.length) {
+    const bracket = tagIsVoidElement && voidElementMode === 'xhtml' ? ' />' : '>';
+    if ('attrs' in node && node.attrs.length === 1) {
```

Comment:
I get that you are trying to be safer by using `x in y`, but this is actually less safe now.

The code previously said:

* If node has a value
* and node.attrs has a value
* and node.attrs.length has a value of 1

Now it says

* I assume node has a value
* If node has a property called attrs, even if it is undefined
* assume it always has a length
* check if the length equals 1

We could change it to this:

`if ('attrs' in node && node?.attrs?.length === 1) {`

But then it's just redundant, and can be simplified:

```suggestion
    if (node?.attrs?.length === 1) {
```

Reply:
After reading through some source code, I've changed this to use the type guards from parse5 (e.g. `defaultTreeAdapter.isElementNode(node)`).

It should be reasonable to assume `node` is a `ChildNode` (especially since `formatNode` is defined and used only within the exported `diffableFormatter`).